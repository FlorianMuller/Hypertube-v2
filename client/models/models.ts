import { AxiosResponse, AxiosError, Canceler } from "axios";

export interface Locale {
  locale: string;
  setLocale: (locale: string) => void;
}

export type Method = "get" | "put" | "post" | "patch" | "delete";
export type Headers = { [key: string]: string } | undefined;
export type Data = object | undefined;

export interface UseApiOption {
  method?: Method;
  headers?: Headers;
  data?: Data;
  hotReload?: boolean;
  validateStatus?: (status: number) => boolean;
}

export interface UseApiReturn<T, E> {
  callApi: (freshData?: Data) => Promise<void>;
  loading: boolean;
  res: AxiosResponse<T>;
  resData: T;
  error: AxiosError<E>;
  setUrl: (url: string) => void;
  setMethod: (method: Method) => void;
  setHeaders: (headers: Headers) => void;
  setData: (data: Data) => void;
  cancelAllRequests: Canceler;
}

export interface ApiAuthResponse {
  validToken: boolean;
  error?: string;
}

export interface AuthInfo {
  username: string;
  password: string;
  [key: string]: string;
}

export interface Comment {
  movieName: string;
  date: number;
  name: string;
  body: string;
  stars: number;
  _id: number;
}

export interface Review {
  id: string;
  name: string;
  date: number;
  stars: number;
  body: string;
}

export interface Filters {
  query: string;
  collections: string[];
  year: number;
  minRating: number;
}

export interface Movie {
  id: string;
  title: string;
  cover: string;
  year: number;
  summary: string;
  genres: string[];
  rating: number;
  runtime: number;
}

export interface ApiSearchReponse {
  movies: Movie[];
  nextPage: boolean;
}

export type CustomSnackbarVariant =
  | "success"
  | "warning"
  | "error"
  | "info"
  | "default";
