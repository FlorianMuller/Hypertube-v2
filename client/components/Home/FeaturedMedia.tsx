import React, { ReactElement } from "react";
import { Skeleton, Rating } from "@material-ui/lab";
import { useHistory } from "react-router-dom";
import { useIntl } from "react-intl";
import {
  Button,
  Typography,
  Chip,
  Paper,
  useMediaQuery,
  useTheme
} from "@material-ui/core";
import Image from "material-ui-image";

import { MovieData } from "../../models/models";

import { getPictureSize } from "./service";
import { useFilmStyles } from "./styles";

import useApi from "../../hooks/useApi";

const FeaturedMedia = (): ReactElement => {
  const classes = useFilmStyles({});
  const history = useHistory();
  const { formatMessage: _t } = useIntl();
  const theme = useTheme();
  const isXs = useMediaQuery(theme.breakpoints.down("sm"));
  const { data: media, loading, error } = (useApi(
    "/search/latest-movie"
  ) as unknown) as MovieData;

  if (error) {
    history.push("/error");
  }

  if (loading) {
    return (
      <Paper className={classes.container}>
        <Typography variant="h4" className={classes.title}>
          {_t({ id: "home.latest_film" })}
        </Typography>
        <div className={classes.posterContainer}>
          <Skeleton variant="rect" height={562.5} width={375} />
        </div>
      </Paper>
    );
  }

  const renderTags = (): ReactElement[] =>
    media.genres?.map((genre: string) => (
      <Chip
        key={genre}
        className={classes.tag}
        label={genre}
        clickable
        color="primary"
      />
    ));

  return (
    <Paper className={classes.container}>
      <Typography variant="h4" className={classes.title}>
        {_t({ id: "home.latest_film" })}
      </Typography>
      <div className={classes.posterContainer}>
        <Image
          animationDuration={500}
          src={media.cover}
          imageStyle={getPictureSize(isXs)}
          style={getPictureSize(isXs)}
          color="rgba(0,0,0,0)"
        />
      </div>
      <div className={classes.infosOverlay}>
        <Typography variant="h3" className={classes.title}>
          {media.title}
        </Typography>
        <Typography variant="h4">{media.year}</Typography>
        <Typography className={classes.descriptionContainer}>
          {media.summary}
        </Typography>
        <Rating readOnly value={media.rating} precision={0.1} />
        <div>{renderTags()}</div>
        <div className={classes.buttonContainer}>
          <Button
            color="primary"
            variant="contained"
            fullWidth
            onClick={(): void => history.push(`/movie/${media.id}`)}
          >
            {_t({ id: "search.film.watch" })}
          </Button>
        </div>
      </div>
    </Paper>
  );
};

export default FeaturedMedia;
