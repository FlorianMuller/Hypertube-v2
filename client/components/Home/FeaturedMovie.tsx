import React, { ReactElement } from "react";
import { Skeleton, Rating } from "@material-ui/lab";
import { useHistory } from "react-router-dom";
import { useIntl } from "react-intl";

import { Button, Typography, Chip, Paper } from "@material-ui/core";
import Image from "material-ui-image";

import { MovieData } from "../../models/models";

import { useFilmStyles } from "./styles";

import useApi from "../../hooks/useApi";

const Film = (): ReactElement => {
  const classes = useFilmStyles({});
  const history = useHistory();
  const { formatMessage: _t } = useIntl();
  const { data: movie, loading, error } = (useApi(
    "/search/latest-movie"
  ) as unknown) as MovieData;

  if (error) {
    return <div>error</div>;
  }

  if (loading) {
    return (
      <Paper className={classes.container}>
        <Typography variant="h4" className={classes.title}>
          {_t({ id: "home.latest_film" })}
        </Typography>
        <div className={classes.posterContainer}>
          <Skeleton variant="rect" height={562.5} width={375} />
        </div>
      </Paper>
    );
  }

  const renderTags = (): ReactElement[] =>
    movie.genres?.map((genre: string) => (
      <Chip
        key={genre}
        className={classes.tag}
        label={genre}
        clickable
        color="primary"
      />
    ));

  return (
    <Paper className={classes.container}>
      <Typography variant="h4" className={classes.title}>
        {_t({ id: "home.latest_film" })}
      </Typography>
      <div className={classes.posterContainer}>
        <Image
          animationDuration={500}
          src={movie.cover}
          imageStyle={{ width: 375, height: 562.5 }}
          style={{ width: 375, height: 562.5 }}
          color="rgba(0,0,0,0)"
        />
      </div>
      <div className={classes.infosOverlay}>
        <Typography variant="h3" className={classes.title}>
          {movie.title}
        </Typography>
        <Typography variant="h4">{movie.year}</Typography>
        <Typography className={classes.descriptionContainer}>
          {movie.summary}
        </Typography>
        <Rating readOnly value={movie.rating} precision={0.1} />
        <div>{renderTags()}</div>
        <div className={classes.buttonContainer}>
          <Button
            color="primary"
            variant="contained"
            fullWidth
            onClick={(): void => history.push(`/movie/${movie.id}`)}
          >
            {_t({ id: "search.film.watch" })}
          </Button>
        </div>
      </div>
    </Paper>
  );
};

export default Film;
