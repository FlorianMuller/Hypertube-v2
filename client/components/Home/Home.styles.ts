import { makeStyles } from "@material-ui/core/styles";

export const useHomeStyles = makeStyles((theme) => ({
  mainPoster: {
    background:
      "linear-gradient(to bottom, transparent 40%, black 80%), url('/public/backgroundImage.jpg')",
    backgroundRepeat: "no-repeat",
    backgroundSize: "cover",
    backgroundPosition: "50% 0%",
    width: "100%",
    height: "100%",
    marginTop: "0",
    display: "flex"
  },
  boxContent: {
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "center",
    width: "100%",
    marginTop: theme.spacing(10)
  }
}));

export const useRecentMoviesStyle = makeStyles(() => ({}));
