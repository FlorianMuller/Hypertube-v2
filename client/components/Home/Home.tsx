import React, { ReactElement } from 'react';

import useApi from '../../hooks/useApi';

import Authentication from '../Authentication/Authentication';

const Home = (): ReactElement => {
  const { data: { validToken }, loading, error } = useApi('http://localhost:8080/API/check-token');

  if (error) {
    return <div>Error</div>;
  }
  if (loading) {
    return <div>Loading</div>;
  }
  if (!validToken) {
    return <Authentication />;
  }

  return (
    <div>
            Home sweet home;
    </div>
  );
};

export default Home;
