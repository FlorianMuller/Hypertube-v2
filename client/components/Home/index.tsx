import React, { ReactElement } from "react";
import { useHistory } from "react-router";
import { Typography, Paper, Hidden } from "@material-ui/core";

import useApi from "../../hooks/useApi";

import SignUp from "../Authentication/SignUp";

import { useHomeStyles } from "./styles";

import Feed from "./Feed";
import RecentVideos from "./RecentMedias";
import FeaturedMedia from "./FeaturedMedia";

const Home = (): ReactElement => {
  const history = useHistory();
  const {
    data: { validToken },
    loading,
    error
  } = useApi("/check-token");
  const classes = useHomeStyles({});

  if (error) {
    history.push("/error");
  }
  if (loading) {
    return <div>Loading</div>;
  }
  if (!validToken) {
    return <SignUp />;
  }

  return (
    <div className={classes.container}>
      <div className={classes.header}>
        <Hidden xsDown>
          <Typography variant="h1">Welcome Julien</Typography>
          <Typography variant="h2">Start browsing videos</Typography>
        </Hidden>
        <Hidden smUp>
          <Typography variant="h4">Welcome Julien</Typography>
          <Typography variant="h5">Start browsing videos</Typography>
        </Hidden>
      </div>
      <div className={classes.mainContent}>
        <div className={classes.feedContainer}>
          <Feed />
        </div>
        <div className={classes.filmContainer}>
          <FeaturedMedia />
        </div>
      </div>
      <div className={classes.subContent}>
        <div className={classes.recentVideosContainer}>
          <RecentVideos />
        </div>
        <div className={classes.dataContainer}>
          <Paper className={classes.infosContainer}>
            <div>User online</div>
          </Paper>
        </div>
      </div>
    </div>
  );
};

export default Home;
