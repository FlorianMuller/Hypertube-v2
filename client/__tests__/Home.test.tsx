import React from "react";
import { StaticRouter } from "react-router-dom";
import EnzymeToJson from "enzyme-to-json";

import Home from "../components/Home/Home";
import { mountWithIntl } from "./helpers/intl-enzyme-test-helper";
import { UseApiReturn, Movie } from "../models/models";

const mockedMoviesList = [
  {
    id: "1",
    title: "The comic movies",
    cover: "path/to/pic",
    year: 2020,
    summary: "The one with Dany Boon you kwow, all the France is gonna laugh",
    genres: ["comedy"],
    rating: 4,
    runtime: 100,
    viewed: false
  },
  {
    id: "2",
    title: "The marvel block buster",
    cover: "path/to/pic",
    year: 2020,
    summary: "You've seen it hundreds time, but here it is, an other one",
    genres: ["action", "adventure"],
    rating: 4,
    runtime: 100,
    viewed: false
  },
  {
    id: "3",
    title: "The cool kid moovie",
    cover: "path/to/pic",
    year: 2020,
    summary: "The one with jokes for the parents",
    genres: ["action", "adventure"],
    rating: 4,
    runtime: 100,
    viewed: false
  },
  {
    id: "4",
    title: "The horror movie",
    cover: "path/to/pic",
    year: 2020,
    summary: "I don't know anything about it, I was too afraid to see it",
    genres: ["horror"],
    rating: 4,
    runtime: 100,
    viewed: false
  }
];

jest.mock("../hooks/useApi", () => (): UseApiReturn<
  { list: Movie[] },
  void
> => ({
  callApi: jest.fn(),
  loading: false,
  res: {
    data: { list: mockedMoviesList },
    status: 200,
    statusText: null,
    headers: null,
    config: null
  },
  resData: { list: mockedMoviesList },
  error: null,
  setUrl: jest.fn(),
  setMethod: jest.fn(),
  setHeaders: jest.fn(),
  setData: jest.fn(),
  cancelAllRequests: jest.fn()
}));

describe("Home", () => {
  it("should render <Home> in english", () => {
    const domNode = mountWithIntl(
      <StaticRouter>
        <Home />
      </StaticRouter>,
      "en"
    );
    expect(EnzymeToJson(domNode)).toMatchSnapshot();
  });

  it("should render <Home> in french", () => {
    const domNode = mountWithIntl(
      <StaticRouter>
        <Home />
      </StaticRouter>,
      "fr"
    );
    expect(EnzymeToJson(domNode)).toMatchSnapshot();
  });
});
