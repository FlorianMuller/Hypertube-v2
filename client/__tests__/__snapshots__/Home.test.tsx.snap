// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render <Home> in english 1`] = `
<StaticRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Home>
      <div
        className="makeStyles-mainPoster-1"
      >
        <div
          className="makeStyles-boxContent-2"
        >
          <MovieThumbnail
            key="1"
            movie={
              Object {
                "cover": "path/to/pic",
                "genres": Array [
                  "comedy",
                ],
                "id": "1",
                "rating": 4,
                "runtime": 100,
                "summary": "The one with Dany Boon you kwow, all the France is gonna laugh",
                "title": "The comic movies",
                "viewed": false,
                "year": 2020,
              }
            }
          >
            <div
              className="makeStyles-thumbnailContainer-4"
              key="1"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <Image
                animationDuration={500}
                aspectRatio={1}
                color="rgba(0,0,0,0)"
                disableError={false}
                disableSpinner={true}
                disableTransition={false}
                errorIcon={
                  <ForwardRef(WithStyles)
                    className="makeStyles-altContainer-14"
                  >
                    <ForwardRef(WithStyles)
                      variant="h4"
                    >
                      The comic movies
                    </ForwardRef(WithStyles)>
                  </ForwardRef(WithStyles)>
                }
                imageStyle={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
                loading={
                  <ForwardRef(WithStyles)
                    size={48}
                  />
                }
                src="path/to/pic"
                style={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,0)",
                      "height": 450,
                      "paddingTop": "calc(1 / 1 * 100%)",
                      "position": "relative",
                      "width": 300,
                    }
                  }
                >
                  <img
                    onError={[Function]}
                    onLoad={[Function]}
                    src="path/to/pic"
                    style={
                      Object {
                        "filterBrightness": 0,
                        "filterSaturate": 20,
                        "height": 450,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "top": 0,
                        "transition": "
        filterBrightness 375ms cubic-bezier(0.4, 0.0, 0.2, 1),
        filterSaturate 500ms cubic-bezier(0.4, 0.0, 0.2, 1),
        opacity 250ms cubic-bezier(0.4, 0.0, 0.2, 1)",
                        "width": 300,
                      }
                    }
                  />
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "height": "100%",
                        "justifyContent": "center",
                        "left": 0,
                        "pointerEvents": "none",
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </Image>
            </div>
          </MovieThumbnail>
          <MovieThumbnail
            key="2"
            movie={
              Object {
                "cover": "path/to/pic",
                "genres": Array [
                  "action",
                  "adventure",
                ],
                "id": "2",
                "rating": 4,
                "runtime": 100,
                "summary": "You've seen it hundreds time, but here it is, an other one",
                "title": "The marvel block buster",
                "viewed": false,
                "year": 2020,
              }
            }
          >
            <div
              className="makeStyles-thumbnailContainer-4"
              key="2"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <Image
                animationDuration={500}
                aspectRatio={1}
                color="rgba(0,0,0,0)"
                disableError={false}
                disableSpinner={true}
                disableTransition={false}
                errorIcon={
                  <ForwardRef(WithStyles)
                    className="makeStyles-altContainer-14"
                  >
                    <ForwardRef(WithStyles)
                      variant="h4"
                    >
                      The marvel block buster
                    </ForwardRef(WithStyles)>
                  </ForwardRef(WithStyles)>
                }
                imageStyle={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
                loading={
                  <ForwardRef(WithStyles)
                    size={48}
                  />
                }
                src="path/to/pic"
                style={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,0)",
                      "height": 450,
                      "paddingTop": "calc(1 / 1 * 100%)",
                      "position": "relative",
                      "width": 300,
                    }
                  }
                >
                  <img
                    onError={[Function]}
                    onLoad={[Function]}
                    src="path/to/pic"
                    style={
                      Object {
                        "filterBrightness": 0,
                        "filterSaturate": 20,
                        "height": 450,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "top": 0,
                        "transition": "
        filterBrightness 375ms cubic-bezier(0.4, 0.0, 0.2, 1),
        filterSaturate 500ms cubic-bezier(0.4, 0.0, 0.2, 1),
        opacity 250ms cubic-bezier(0.4, 0.0, 0.2, 1)",
                        "width": 300,
                      }
                    }
                  />
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "height": "100%",
                        "justifyContent": "center",
                        "left": 0,
                        "pointerEvents": "none",
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </Image>
            </div>
          </MovieThumbnail>
          <MovieThumbnail
            key="3"
            movie={
              Object {
                "cover": "path/to/pic",
                "genres": Array [
                  "action",
                  "adventure",
                ],
                "id": "3",
                "rating": 4,
                "runtime": 100,
                "summary": "The one with jokes for the parents",
                "title": "The cool kid moovie",
                "viewed": false,
                "year": 2020,
              }
            }
          >
            <div
              className="makeStyles-thumbnailContainer-4"
              key="3"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <Image
                animationDuration={500}
                aspectRatio={1}
                color="rgba(0,0,0,0)"
                disableError={false}
                disableSpinner={true}
                disableTransition={false}
                errorIcon={
                  <ForwardRef(WithStyles)
                    className="makeStyles-altContainer-14"
                  >
                    <ForwardRef(WithStyles)
                      variant="h4"
                    >
                      The cool kid moovie
                    </ForwardRef(WithStyles)>
                  </ForwardRef(WithStyles)>
                }
                imageStyle={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
                loading={
                  <ForwardRef(WithStyles)
                    size={48}
                  />
                }
                src="path/to/pic"
                style={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,0)",
                      "height": 450,
                      "paddingTop": "calc(1 / 1 * 100%)",
                      "position": "relative",
                      "width": 300,
                    }
                  }
                >
                  <img
                    onError={[Function]}
                    onLoad={[Function]}
                    src="path/to/pic"
                    style={
                      Object {
                        "filterBrightness": 0,
                        "filterSaturate": 20,
                        "height": 450,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "top": 0,
                        "transition": "
        filterBrightness 375ms cubic-bezier(0.4, 0.0, 0.2, 1),
        filterSaturate 500ms cubic-bezier(0.4, 0.0, 0.2, 1),
        opacity 250ms cubic-bezier(0.4, 0.0, 0.2, 1)",
                        "width": 300,
                      }
                    }
                  />
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "height": "100%",
                        "justifyContent": "center",
                        "left": 0,
                        "pointerEvents": "none",
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </Image>
            </div>
          </MovieThumbnail>
          <MovieThumbnail
            key="4"
            movie={
              Object {
                "cover": "path/to/pic",
                "genres": Array [
                  "horror",
                ],
                "id": "4",
                "rating": 4,
                "runtime": 100,
                "summary": "I don't know anything about it, I was too afraid to see it",
                "title": "The horror movie",
                "viewed": false,
                "year": 2020,
              }
            }
          >
            <div
              className="makeStyles-thumbnailContainer-4"
              key="4"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <Image
                animationDuration={500}
                aspectRatio={1}
                color="rgba(0,0,0,0)"
                disableError={false}
                disableSpinner={true}
                disableTransition={false}
                errorIcon={
                  <ForwardRef(WithStyles)
                    className="makeStyles-altContainer-14"
                  >
                    <ForwardRef(WithStyles)
                      variant="h4"
                    >
                      The horror movie
                    </ForwardRef(WithStyles)>
                  </ForwardRef(WithStyles)>
                }
                imageStyle={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
                loading={
                  <ForwardRef(WithStyles)
                    size={48}
                  />
                }
                src="path/to/pic"
                style={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,0)",
                      "height": 450,
                      "paddingTop": "calc(1 / 1 * 100%)",
                      "position": "relative",
                      "width": 300,
                    }
                  }
                >
                  <img
                    onError={[Function]}
                    onLoad={[Function]}
                    src="path/to/pic"
                    style={
                      Object {
                        "filterBrightness": 0,
                        "filterSaturate": 20,
                        "height": 450,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "top": 0,
                        "transition": "
        filterBrightness 375ms cubic-bezier(0.4, 0.0, 0.2, 1),
        filterSaturate 500ms cubic-bezier(0.4, 0.0, 0.2, 1),
        opacity 250ms cubic-bezier(0.4, 0.0, 0.2, 1)",
                        "width": 300,
                      }
                    }
                  />
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "height": "100%",
                        "justifyContent": "center",
                        "left": 0,
                        "pointerEvents": "none",
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </Image>
            </div>
          </MovieThumbnail>
        </div>
      </div>
    </Home>
  </Router>
</StaticRouter>
`;

exports[`Home should render <Home> in french 1`] = `
<StaticRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Home>
      <div
        className="makeStyles-mainPoster-1"
      >
        <div
          className="makeStyles-boxContent-2"
        >
          <MovieThumbnail
            key="1"
            movie={
              Object {
                "cover": "path/to/pic",
                "genres": Array [
                  "comedy",
                ],
                "id": "1",
                "rating": 4,
                "runtime": 100,
                "summary": "The one with Dany Boon you kwow, all the France is gonna laugh",
                "title": "The comic movies",
                "viewed": false,
                "year": 2020,
              }
            }
          >
            <div
              className="makeStyles-thumbnailContainer-4"
              key="1"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <Image
                animationDuration={500}
                aspectRatio={1}
                color="rgba(0,0,0,0)"
                disableError={false}
                disableSpinner={true}
                disableTransition={false}
                errorIcon={
                  <ForwardRef(WithStyles)
                    className="makeStyles-altContainer-14"
                  >
                    <ForwardRef(WithStyles)
                      variant="h4"
                    >
                      The comic movies
                    </ForwardRef(WithStyles)>
                  </ForwardRef(WithStyles)>
                }
                imageStyle={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
                loading={
                  <ForwardRef(WithStyles)
                    size={48}
                  />
                }
                src="path/to/pic"
                style={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,0)",
                      "height": 450,
                      "paddingTop": "calc(1 / 1 * 100%)",
                      "position": "relative",
                      "width": 300,
                    }
                  }
                >
                  <img
                    onError={[Function]}
                    onLoad={[Function]}
                    src="path/to/pic"
                    style={
                      Object {
                        "filterBrightness": 0,
                        "filterSaturate": 20,
                        "height": 450,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "top": 0,
                        "transition": "
        filterBrightness 375ms cubic-bezier(0.4, 0.0, 0.2, 1),
        filterSaturate 500ms cubic-bezier(0.4, 0.0, 0.2, 1),
        opacity 250ms cubic-bezier(0.4, 0.0, 0.2, 1)",
                        "width": 300,
                      }
                    }
                  />
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "height": "100%",
                        "justifyContent": "center",
                        "left": 0,
                        "pointerEvents": "none",
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </Image>
            </div>
          </MovieThumbnail>
          <MovieThumbnail
            key="2"
            movie={
              Object {
                "cover": "path/to/pic",
                "genres": Array [
                  "action",
                  "adventure",
                ],
                "id": "2",
                "rating": 4,
                "runtime": 100,
                "summary": "You've seen it hundreds time, but here it is, an other one",
                "title": "The marvel block buster",
                "viewed": false,
                "year": 2020,
              }
            }
          >
            <div
              className="makeStyles-thumbnailContainer-4"
              key="2"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <Image
                animationDuration={500}
                aspectRatio={1}
                color="rgba(0,0,0,0)"
                disableError={false}
                disableSpinner={true}
                disableTransition={false}
                errorIcon={
                  <ForwardRef(WithStyles)
                    className="makeStyles-altContainer-14"
                  >
                    <ForwardRef(WithStyles)
                      variant="h4"
                    >
                      The marvel block buster
                    </ForwardRef(WithStyles)>
                  </ForwardRef(WithStyles)>
                }
                imageStyle={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
                loading={
                  <ForwardRef(WithStyles)
                    size={48}
                  />
                }
                src="path/to/pic"
                style={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,0)",
                      "height": 450,
                      "paddingTop": "calc(1 / 1 * 100%)",
                      "position": "relative",
                      "width": 300,
                    }
                  }
                >
                  <img
                    onError={[Function]}
                    onLoad={[Function]}
                    src="path/to/pic"
                    style={
                      Object {
                        "filterBrightness": 0,
                        "filterSaturate": 20,
                        "height": 450,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "top": 0,
                        "transition": "
        filterBrightness 375ms cubic-bezier(0.4, 0.0, 0.2, 1),
        filterSaturate 500ms cubic-bezier(0.4, 0.0, 0.2, 1),
        opacity 250ms cubic-bezier(0.4, 0.0, 0.2, 1)",
                        "width": 300,
                      }
                    }
                  />
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "height": "100%",
                        "justifyContent": "center",
                        "left": 0,
                        "pointerEvents": "none",
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </Image>
            </div>
          </MovieThumbnail>
          <MovieThumbnail
            key="3"
            movie={
              Object {
                "cover": "path/to/pic",
                "genres": Array [
                  "action",
                  "adventure",
                ],
                "id": "3",
                "rating": 4,
                "runtime": 100,
                "summary": "The one with jokes for the parents",
                "title": "The cool kid moovie",
                "viewed": false,
                "year": 2020,
              }
            }
          >
            <div
              className="makeStyles-thumbnailContainer-4"
              key="3"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <Image
                animationDuration={500}
                aspectRatio={1}
                color="rgba(0,0,0,0)"
                disableError={false}
                disableSpinner={true}
                disableTransition={false}
                errorIcon={
                  <ForwardRef(WithStyles)
                    className="makeStyles-altContainer-14"
                  >
                    <ForwardRef(WithStyles)
                      variant="h4"
                    >
                      The cool kid moovie
                    </ForwardRef(WithStyles)>
                  </ForwardRef(WithStyles)>
                }
                imageStyle={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
                loading={
                  <ForwardRef(WithStyles)
                    size={48}
                  />
                }
                src="path/to/pic"
                style={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,0)",
                      "height": 450,
                      "paddingTop": "calc(1 / 1 * 100%)",
                      "position": "relative",
                      "width": 300,
                    }
                  }
                >
                  <img
                    onError={[Function]}
                    onLoad={[Function]}
                    src="path/to/pic"
                    style={
                      Object {
                        "filterBrightness": 0,
                        "filterSaturate": 20,
                        "height": 450,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "top": 0,
                        "transition": "
        filterBrightness 375ms cubic-bezier(0.4, 0.0, 0.2, 1),
        filterSaturate 500ms cubic-bezier(0.4, 0.0, 0.2, 1),
        opacity 250ms cubic-bezier(0.4, 0.0, 0.2, 1)",
                        "width": 300,
                      }
                    }
                  />
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "height": "100%",
                        "justifyContent": "center",
                        "left": 0,
                        "pointerEvents": "none",
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </Image>
            </div>
          </MovieThumbnail>
          <MovieThumbnail
            key="4"
            movie={
              Object {
                "cover": "path/to/pic",
                "genres": Array [
                  "horror",
                ],
                "id": "4",
                "rating": 4,
                "runtime": 100,
                "summary": "I don't know anything about it, I was too afraid to see it",
                "title": "The horror movie",
                "viewed": false,
                "year": 2020,
              }
            }
          >
            <div
              className="makeStyles-thumbnailContainer-4"
              key="4"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <Image
                animationDuration={500}
                aspectRatio={1}
                color="rgba(0,0,0,0)"
                disableError={false}
                disableSpinner={true}
                disableTransition={false}
                errorIcon={
                  <ForwardRef(WithStyles)
                    className="makeStyles-altContainer-14"
                  >
                    <ForwardRef(WithStyles)
                      variant="h4"
                    >
                      The horror movie
                    </ForwardRef(WithStyles)>
                  </ForwardRef(WithStyles)>
                }
                imageStyle={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
                loading={
                  <ForwardRef(WithStyles)
                    size={48}
                  />
                }
                src="path/to/pic"
                style={
                  Object {
                    "height": 450,
                    "width": 300,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,0)",
                      "height": 450,
                      "paddingTop": "calc(1 / 1 * 100%)",
                      "position": "relative",
                      "width": 300,
                    }
                  }
                >
                  <img
                    onError={[Function]}
                    onLoad={[Function]}
                    src="path/to/pic"
                    style={
                      Object {
                        "filterBrightness": 0,
                        "filterSaturate": 20,
                        "height": 450,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "top": 0,
                        "transition": "
        filterBrightness 375ms cubic-bezier(0.4, 0.0, 0.2, 1),
        filterSaturate 500ms cubic-bezier(0.4, 0.0, 0.2, 1),
        opacity 250ms cubic-bezier(0.4, 0.0, 0.2, 1)",
                        "width": 300,
                      }
                    }
                  />
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "height": "100%",
                        "justifyContent": "center",
                        "left": 0,
                        "pointerEvents": "none",
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </Image>
            </div>
          </MovieThumbnail>
        </div>
      </div>
    </Home>
  </Router>
</StaticRouter>
`;
